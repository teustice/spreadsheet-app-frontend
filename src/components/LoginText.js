import React, {Component} from 'react';
import { connect } from 'react-redux';

class LoginText extends Component {
    render() {
        return (
            <div className={"login-text"} >
                {this.props.currentUser &&
                    <p>Hello, {this.props.currentUser.email}</p>
                }
            </div>
        );
    }
}

const mapStateToProps = state => ({
 ...state
})

const mapDispatchToProps = dispatch => ({
})

export default connect(mapStateToProps, mapDispatchToProps)(LoginText);
