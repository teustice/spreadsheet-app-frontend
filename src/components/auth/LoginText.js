import React, {Component} from 'react';
import PropTypes from 'prop-types'
import { connect } from 'react-redux';

export class LoginText extends Component {
    render() {
        return (
            <article className={"login-text"} >
                {this.props.currentUser &&
                  <div>
                    <p>Hello, {this.props.currentUser.email}</p>
                  </div>
                }
            </article>
        );
    }
}

LoginText.propTypes = {
  currentUser: PropTypes.object
}

const mapStateToProps = state => ({
 ...state
})

const mapDispatchToProps = dispatch => ({
})

export default connect(mapStateToProps, mapDispatchToProps)(LoginText);
