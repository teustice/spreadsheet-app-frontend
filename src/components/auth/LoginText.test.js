import React from "react";
import {configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import {LoginText} from "./LoginText";

configure({ adapter: new Adapter() });

describe('<LoginText />', () => {
  describe('render()', () => {

    test('renders the component', () => {
      expect(toJson(<LoginText />)).toMatchSnapshot();
    });

    test('renders text when a user is logged in', () => {
      const wrapper = shallow(<LoginText currentUser={{email: 'test@test.com'}} />);
      expect(wrapper.find('p').text()).toEqual(`Hello, test@test.com`);
    });


    test('It displays nothing without a logged in user', () => {
      const wrapper = shallow(<LoginText />);
      expect(wrapper.find('p').exists()).toEqual(false);
    });
  });
});
