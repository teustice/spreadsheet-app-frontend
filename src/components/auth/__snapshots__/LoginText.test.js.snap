// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginText /> render() renders the component 1`] = `undefined`;
